docs/src/chapters/testy.adoc

```diff
@@ -1,12 +1,262 @@
+lub czy wpisany klucz może być kluczem publicznym.
+
+.Katalog common
+image::../images/tests_chapter/common_folder.png[]
+
+* helper- W tym folderze umieszczona została implementacja pomocnicza. Stworzona tutaj została np. metoda
+_resetSettings_, odpowiadająca za powrót do domyślnych właściwości, czy też implementacja klasy _BaseIntegrationTest_
+wykorzystywane w testach integracyjnych.
+
+.Katalog helper
+image::../images/tests_chapter/helper_folder.png[]
+
+* notification- W tym katalogu umieszczono testy, które związane były z powiadomieniami. Sprawdzano np. czy
+powiadomienie o danym _id_ zostanie prawidłowo zwrócone oraz czy jego typ jest zgodny z oczekiwaniami.
+
+.Katalog notification
+image::../images/tests_chapter/notification_folder.png[]
+
+* payment- Tak jak sama nazwa wskazuje, w poniższych folderach umieszczone zostały testy związane z płatnościami.
+Miały one na celu np. sprawdzić, czy status płatności zachowuje się zgodnie z oczekiwaniami.
+
+.Katalog payments
+image::../images/tests_chapter/payments_folder.png[]
+
+* transaction- Katalog ten zawiera 4 pliki z testami. Ich celem było sprawdzenie, czy funkcjonalności związane z
+transakcjami działają prawidłowo.
+
+.Katalog transaction
+image::../images/tests_chapter/transaction_folder.png[]
+
+* wallet- W tej paczce znalazły się testy sprawdzające funkcjonalności związane z portfelem.
+
+.Katalog wallet
+image::../images/tests_chapter/wallet_folder.png[]
+
+Poniżej zaprezentowany został przykładowy test jednostkowy. Oczekujemy w nim, że po prawidłowym dodaniu administratora
+zwrócony zostanie kod odpowiedzi HTTP 201(_utworzono_).
+
+.Test jednostkowy z pliku AdminResourceTest.java
+[source, java]
+----
+include::../../../webservice/src/test/java/pl/edu/pjatk/lnpayments/webservice/admin/resource/AdminResourceTest.java[lines=30..38]
+----
+Kolejny fragment kodu zawiera natomiast test integracyjny. Służy on do sprawdzenia, czy w przypadku wprowadzenia
+nieprawidłowych danych zwrócony zostanie kod odpowiedzi HTTP _złe zapytanie_.
+
+.Test integracyjny z pliku AdminResourceIntegrationTest.java
+[source, java]
+----
+include::../../../webservice/src/test/java/pl/edu/pjatk/lnpayments/webservice/admin/resource/AdminResourceIntegrationTest.java[lines=82..89]
+----
+
+Do tworzenia testów wykorzystane zostały biblioteki _JUnit_ oraz _Mockito_. Warto również wspomnieć o tym, że wszystkie
+klasy z testami integracyjnymi korzystają z klasy _BaseIntegrationTest_.
+
+Na poniższym zrzucie ekranu został załączony raport dotyczący pokrycia testami kodu związanego z częścią serwerową
+aplikacji. Wynik na poziomie 91% świadczy o solidnym podejściu do testowania stworzonego kodu. Niektóre klasy zostały
+pokryte testami nawet w 100%, co wymagało poświęcenia trochę dodatkowego czasu.
+
+.Raport z pokrycia testami kodu części backendowej
+image::../images/tests_chapter/test_coverage_backend.png[]
```

Comment:
Prosiłem żeby pokazać raport z jacoco, ten zawiera wszystkie takie same dane/jest w czymś lepszy?

Reply:
to jest ten raport z jacoco ( plik jacoco.exec) tylko wyświetlony przez Intellij ( run -> show coverage data)

Reply:
pokazuje inne dane, dlatego pytam. Porównaj sobie z tym co wypluje mvm verify -Ptest-coverage w target/site/jacoco/index.html

Reply:
faktycznie, w index.html jest coverage 95%. Problem w tym, że tamta tabela zawiera listę wszystkich plików, więc jeśli chcemy wykorzystać tamten raport to trzeba się bawić w mistrza painta i wyciąć nazwę kolumn z tabeli + ostatni wiersz total z podsumowaniem, bo nie wrzucimy na cała stronę ssa

Reply:
dobra, to zrób jak mówiłem na daili, 2-3 zdania o tym że mieliśmy raporty z jacoco, coś tam napisać o różnicach